test/test_rack_handler.rb

```diff
@@ -2,16 +2,22 @@
 
 require "rack"
 
-if Rack::RELEASE < '3'
-
-  require "rack/handler/puma"
+module TestRackUp
+  if Rack::RELEASE < '3'
+    require "rack/handler/puma"
+    RACK_HANDLER_MOD = ::Rack::Handler
+  else
+    require "rackup"
+    require "rack/handler/puma"
+    RACK_HANDLER_MOD = ::Rackup::Handler
```

Comment:
Maybe we could just test `Puma::RackHandler` directly for most of the tests instead?

Reply:
Two methods are called in `RACK_HANDLER_MOD`, `register` and `get`.  If we want to verify that things work with Rack/Rackup, it would be good to continue doing so...